policy/modules/services/spamassassin.te

```diff
@@ -95,6 +98,16 @@ files_tmpfs_file(spamd_tmpfs_t)
 type spamd_var_lib_t;
 files_type(spamd_var_lib_t)
 
+type spamd_update_t;
+type spamd_update_exec_t;
+application_domain(spamd_update_t, spamd_update_exec_t)
+role spamd_update_roles types spamd_update_t;
+
+type spamd_update_gpg_t;
```

Comment:
Last year @etbe argued that this was not necessary, which is why the gpg use was removed, please explain the need.

Reply:
gpg is needed to verify the signatures on downloaded spamassassin rules !

without being able to run gpg (with a domain transition **BUT** without being able to access gpg_secret_t files and dirs), spamassassin is not able to verify the signature of new rules downloaded from the Internet and therefore the update fails, as it won't install anything without being able to verify the signature !

gpg signature verification is vital to update both spamassassin classification rules and smartctl drive database (see other update), even though they are not executable or binary files, it's always a good idea to verify that the downloaded files are authentic and genuine (malicious files can still cause DoS and other service disruptions) !